### PR TITLE
Google Cloud BOM 0.167.0 and other related dependencies

### PR DESCRIPTION
**Pull request by suztomo**
# Google Cloud BOM 0.167.0

https://github.com/googleapis/java-cloud-bom/releases/tag/v0.167.0

# Libraries in the shared dependencies BOM 2.7.0

https://github.com/googleapis/java-shared-dependencies/blob/v2.7.0/first-party-dependencies/pom.xml

```
  <properties>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    <site.installationModule>${project.artifactId}</site.installationModule>
    <grpc.version>1.44.0</grpc.version>
    <gax.version>2.11.0</gax.version>
    <grpc-gcp.version>1.1.0</grpc-gcp.version>
    <guava.version>31.0.1-jre</guava.version>
    <protobuf.version>3.19.3</protobuf.version>
    <google.api-common.version>2.1.3</google.api-common.version>
    <google.common-protos.version>2.7.2</google.common-protos.version>
    <google.core.version>2.4.0</google.core.version>
    <google.auth.version>1.4.0</google.auth.version>
    <google.http-client.version>1.41.2</google.http-client.version>
    <google.oauth-client.version>1.33.0</google.oauth-client.version>
    <google.api-client.version>1.33.1</google.api-client.version>
    <iam.version>1.2.1</iam.version>
  </properties>
```

### CODE REVIEW

File: boms/cloud-oss-bom/pom.xml [hunk 1]
```diff
@@ -48,19 +48,19 @@
...
     <!-- We don't use gax-bom because it includes the artifacts with 'testlib' classifier.
         When updating gax.version, update gax.httpjson.version too. -->
-    <gax.version>2.8.1</gax.version>
-    <gax.httpjson.version>0.93.1</gax.httpjson.version>
-    <auth.version>1.3.0</auth.version>
-    <api-common.version>2.1.2</api-common.version>
-    <common.protos.version>2.7.1</common.protos.version>
-    <iam.protos.version>1.2.0</iam.protos.version>
+    <gax.version>2.11.0</gax.version>
+    <gax.httpjson.version>0.96.0</gax.httpjson.version>
```
Comment (suztomo) [inline boms/cloud-oss-bom/pom.xml:59]: Gax-bom 2.11.0 has gax-httpjson 0.96.0.
https://search.maven.org/artifact/com.google.api/gax-bom/2.11.0/pom

**Comment by suztomo** [comment]
```
Error:  testForNewLinkageErrors(com.google.cloud.MaximumLinkageErrorsTest)  Time elapsed: 219.169 s  <<< ERROR!
java.lang.NullPointerException
	at com.google.cloud.MaximumLinkageErrorsTest.testForNewLinkageErrors(MaximumLinkageErrorsTest.java:74)
```

File: dependencies/src/main/resources/linkage-checker-exclusion-default.xml [hunk 2]
```diff
@@ -425,4 +424,25 @@
...
+      Google Cloud libraries do not use SMTP protocols. There are discrepancy in
+      javax.mail:mail:1.4.3 com.sun.mail:javax.mail:1.6.2 in dependency graph.
+    </Reason>
+  </LinkageError>
+  <LinkageError>
+    <Source>
+      <Package name="com.ctc.wstx.shaded" />
+    </Source>
+    <Target>
+      <Class name="com.ctc.wstx.shaded.msv_core.driver.textui.Driver" />
```
Comment (suztomo) [inline dependencies/src/main/resources/linkage-checker-exclusion-default.xml:441]:
```
Class com.ctc.wstx.shaded.msv_core.driver.textui.Driver is not found;
  referenced by 1 class file
    com.ctc.wstx.shaded.msv_core.driver.textui.ReportErrorHandler (com.fasterxml.woodstox:woodstox-core:6.2.6)
```

File: boms/integration-tests/pom.xml [hunk 3]
```diff
@@ -39,6 +39,7 @@
             <!-- To avoid ForkedBooter loading issue -->
             <useSystemClassLoader>false</useSystemClassLoader>
             <argLine>-Xms128m -Xmx2048m</argLine>
+            <trimStackTrace>false</trimStackTrace>
```
Comment (suztomo) [inline boms/integration-tests/pom.xml:42]: This gives better error messages to identify the cause of exceptions.

**Review by mpeddada1 — APPROVED** [review]

**Pull request merged**